File: 1232-check-if-it-is-a-straight-line/1232-check-if-it-is-a-straight-line.cpp
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {
        int diffX=coordinates[1][0]-coordinates[0][0],
            diffY=coordinates[1][1]-coordinates[0][1];
        
        for(int i=2;i<coordinates.size();i++)
        {
            int currDiffX=coordinates[i][0]-coordinates[0][0];
            int currDiffY=coordinates[i][1]-coordinates[0][1];
            
            if(currDiffY*diffX!=currDiffX*diffY)
            {
                return false;
            }
        }
        return true;
    }
};

File: 1232-check-if-it-is-a-straight-line/1232-check-if-it-is-a-straight-line.java
class Solution {
    // (y2 - y1) / (x2 - x1) = (y3 - y2) / (x3 - x2)
    public boolean checkStraightLine(int[][] coordinates) {
        if(coordinates.length==2)
            return true;
        int y = coordinates[1][1] - coordinates[0][1];
        int x = coordinates[1][0] - coordinates[0][0];
        for(int i=1;i<coordinates.length;i++)
        {
            int yi = coordinates[i][1] - coordinates[i-1][1];
            int xi = coordinates[i][0] - coordinates[i-1][0];

            if(xi*y != x*yi)
                return false;
        }
        return true;
    }
}